packages/grid/x-data-grid/src/hooks/features/export/useGridPrintExport.tsx

```diff
@@ -84,18 +93,37 @@ export const useGridPrintExport = (
           newColumnVisibilityModel[column.field] = exportedColumnFields.includes(column.field);
         });
 
+        if (includeCheckboxes) {
+          newColumnVisibilityModel[GRID_CHECKBOX_SELECTION_COL_DEF.field] = true;
+        }
+
         apiRef.current.setColumnVisibilityModel(newColumnVisibilityModel);
         resolve();
       }),
     [apiRef],
   );
 
+  const updateGridRowsForPrint = React.useCallback(
+    (rowsToExport?: (params: GridPrintGetRowsToExportParams<GridApiCommunity>) => GridRowId[]) =>
+      new Promise<void>((resolve) => {
```

Comment:
Is there a reason for `updateGridRowsForPrint` to be asynchronous?

Reply:
The reason I did it like that is because the `updateGridColumnsForPrint` function was setup that way and had these comments above it: `// Returns a promise because updateColumns triggers state update and
  // the new state needs to be in place before the grid can be sized correctly`

So I did it the same to be consistent - not sure if its necessary or not. I just tested removing the promise and it seems to function the same but I don't think I know enough of the context around this to understand if that would mess anything else up or not.

Reply:
I think it's unnecessary, you can remove it 🙂